merlin/models/tf/transformers/block.py

```diff
@@ -0,0 +1,557 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import inspect
+from typing import Optional, Union
+
+import tensorflow as tf
+import transformers
+from transformers import (
+    AlbertConfig,
+    BertConfig,
+    GPT2Config,
+    PretrainedConfig,
+    RobertaConfig,
+    TFPreTrainedModel,
+    XLNetConfig,
+)
+
+from merlin.models.tf.core import combinators
+from merlin.models.tf.core.base import Block, block_registry
+from merlin.models.tf.transformers.transforms import PrepareTransformerInputs
+from merlin.models.tf.utils.tf_utils import (
+    maybe_deserialize_keras_objects,
+    maybe_serialize_keras_objects,
+)
+
+TransformerBody = Union[TFPreTrainedModel, PretrainedConfig, tf.keras.layers.Layer]
+
+
+def get_tf_main_layer(hf_model):
+    """
+    Extract serializable custom keras layer `TF*MainLayer` from the HuggingFace model
+    """
+    main_layer = [v for _, v in hf_model.__dict__.items() if isinstance(v, tf.keras.layers.Layer)][
+        0
+    ]
+    return main_layer
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class TransformerBlock(Block):
+    """
+    Class to support HF Transformers for session-based and sequential-based recommendation models.
+    Parameters
+    ----------
+    transformer: TransformerBody
+        The T4RecConfig, The pre-trained HF model or the custom keras layer TF*MainLayer,
```

Comment:
Is `T4RecConfig` valid here? Or can that part be removed

Reply:
I removed the mention of `T4RecConfig`.